content/en/database_monitoring/setup_oracle/selfhosted.md

```diff
@@ -0,0 +1,208 @@
+        4294967295, TO_NUMBER(NULL), 4294967294, TO_NUMBER(NULL), 4294967293, TO_NUMBER(NULL), 4294967292, TO_NUMBER(NULL),
+        4294967291, TO_NUMBER(NULL), BITAND(s.ksusefblocker, 2147418112) / 65536
+    ) as final_blocking_instance,
+    DECODE(s.ksusefblocker,
+        4294967295, TO_NUMBER(NULL), 4294967294, TO_NUMBER(NULL), 4294967293, TO_NUMBER(NULL), 4294967292, TO_NUMBER(NULL),
+        4294967291, TO_NUMBER(NULL), BITAND(s.ksusefblocker, 65535)
+    ) as final_blocking_session,
+    DECODE(w.kslwtinwait,
+        1, 'WAITING', decode(bitand(w.kslwtflags, 256), 0, 'WAITED UNKNOWN TIME',
+        decode(round(w.kslwtstime / 10000), 0, 'WAITED SHORT TIME', 'WAITED KNOWN TIME'))
+    ) as STATE,
+    e.kslednam as event,
+    e.ksledclass as wait_class,
+    w.kslwtstime as wait_time_micro,
+    c.name as pdb_name,
+    sq.sql_text as sql_text,
+    sq.sql_fulltext as sql_fulltext,
+    sq_prev.sql_fulltext as prev_sql_fulltext,
+    comm.command_name
+FROM
+  x$ksuse s,
+  x$kslwt w,
+  x$ksled e,
+  v$sqlstats sq,
+  v$sqlstats sq_prev,
+  v$containers c,
+  v$sqlcommand comm
+WHERE
+  BITAND(s.ksspaflg, 1) != 0
+  AND BITAND(s.ksuseflg, 1) != 0
+  AND s.inst_id = USERENV('Instance')
+  AND s.indx = w.kslwtsid
+  AND w.kslwtevt = e.indx
+  AND s.ksusesqi = sq.sql_id(+)
+  AND s.ksusesph = sq.plan_hash_value(+)
+  AND s.ksusepsi = sq_prev.sql_id(+)
+  AND s.ksusepha = sq_prev.plan_hash_value(+)
+  AND s.con_id = c.con_id(+)
+  AND s.ksuudoct = comm.command_type(+)
+;
+
+GRANT SELECT ON dd_session TO c##datadog ;
+```
+
+Grant additional permissions if you wish to use the basic oracle integration as well:
+
+```SQL
+GRANT CREATE SESSION TO c##datadog CONTAINER=ALL;
+Grant select any dictionary to c##datadog container=all;
+GRANT SELECT ON GV_$PROCESS TO c##datadog CONTAINER=ALL;
+GRANT SELECT ON gv_$sysmetric TO c##datadog CONTAINER=ALL;
+```
+
+## Install the Agent
+
+To start collecting Oracle telemetry, first [install the Datadog Agent][1]. In order to install the correct version to participate in the beta, please contact your Customer Success Manager.
+
+Create the Oracle Agent conf file `/etc/datadog-agent/conf.d/oracle-dbm.d/conf.yaml`. See the [sample conf file][2] for all available configuration options.
+
+```yaml
+init_config:
+instances:
+  - dbm: true
+    server: '<HOSTNAME>:<SQL_PORT>'
+    service_name: "<SERVICE_NAME>" # The Oracle CDB service name
+    username: 'c##datadog'
+    password: '<PASSWORD>'
+    tags:  # Optional
+      - 'service:<CUSTOM_SERVICE>'
+      - 'env:<CUSTOM_ENV>'
+```
+
+Use the `service` and `env` tags to link your database telemetry to other telemetry through a common tagging scheme. See [Unified Service Tagging][3] on how these tags are used throughout Datadog.
+
+Once all Agent configuration is complete, [restart the Datadog Agent][4].
+
+To install the basic Oracle integration, you should create an additional Oracle Agent conf file at `/etc/datadog-agent/conf.d/oracle.d/conf.yaml`
```

Comment:
```suggestion
To install the basic Oracle integration, you should create an additional Oracle Agent conf file at `/etc/datadog-agent/conf.d/oracle.d/conf.yaml`.
```

Reply:
I'm not too familiar with our Oracle integration(s). Do we have more than one? If not, we can probably remove the word "basic".

Reply:
yeah there is a regular oracle integration and this new one that we have written (eventually they will be merged), and they can be thought of as different feature sets